nncf/common/graph/utils.py

```diff
@@ -132,3 +137,79 @@ def get_reduction_axes(
     for channel_axis in sorted(channel_axes, reverse=True):
         del reduction_axes[channel_axis]
     return tuple(reduction_axes)
+
+
+def get_weight_shape_legacy(layer_attributes: WeightedLayerAttributes) -> List[int]:
+    """
+    Returns hard-coded weights shape layout for the given layer attributes.
+    Applicable only for eager PyTorch and Tensorflow models.
+
+
+    :param layer_attributes: Layer attributes of NNCFNode.
+    :return: Weights shape layout.
+    """
+    if isinstance(layer_attributes, LinearLayerAttributes):
+        return [layer_attributes.out_features, layer_attributes.in_features]
+
+    if isinstance(layer_attributes, ConvolutionLayerAttributes):
+        if not layer_attributes.transpose:
+            return [
+                layer_attributes.out_channels,
+                layer_attributes.in_channels // layer_attributes.groups,
+                *layer_attributes.kernel_size,
+            ]
+        return [
+            layer_attributes.in_channels,
+            layer_attributes.out_channels // layer_attributes.groups,
+            *layer_attributes.kernel_size,
+        ]
+
+    if isinstance(layer_attributes, GroupNormLayerAttributes):
+        return [layer_attributes.num_channels]
+
+    assert isinstance(layer_attributes, GenericWeightedLayerAttributes)
+    return layer_attributes.weight_shape
+
+
+def get_target_dim_for_compression_legacy(layer_attributes: WeightedLayerAttributes) -> int:
+    """
+    Returns hard-coded target dim for compression for the given layer attributes.
+    Applicable only for eager PyTorch and Tensorflow models.
+
+    :param layer_attributes: Layer attributes of NNCFNode.
```

Comment:
Sorry, didn't see it the first time. Please update all the descriptions accordingly 
```suggestion
    :param layer_attributes: Layer attributes of a NNCFNode.
```

Reply:
done